client/my-sites/subscribers/main.tsx

```diff
@@ -181,26 +181,25 @@ const SubscribersPage = ( {
 			sortTermChanged={ sortTermChanged }
 		>
 			<QueryMembershipsSettings siteId={ siteId ?? 0 } source="calypso" />
-			<Main
-				wideLayout
-				className={ `subscribers${
-					isEnabled( 'subscribers-dataviews' ) ? ' subscribers--dataviews' : ''
-				}` }
-			>
+			<Main wideLayout className="subscribers">
 				<DocumentHead title={ translate( 'Subscribers' ) } />
 
-				<SubscribersHeader
-					selectedSiteId={ selectedSite?.ID }
-					disableCta={ isUnverified || isStagingSite }
-				/>
+				{ ! isEnabled( 'subscribers-dataviews' ) && (
+					<SubscribersHeader
+						selectedSiteId={ selectedSite?.ID }
+						disableCta={ isUnverified || isStagingSite }
+					/>
+				) }
 				<SubscriberValidationGate siteId={ siteId }>
 					{ isEnabled( 'subscribers-dataviews' ) ? (
 						// Your new dataviews component
 						<SubscriberDataViews
-							siteId={ siteId }
+							siteId={ selectedSite?.ID }
```

Comment:
Type fix related to the header.

Reply:
Judging by the way `siteId` is being used here I think the better option is probably to change siteId to be

```
const siteId = selectedSite?.ID || 0;
```

That should satisfy your typing and allow us to unify the other places here that are using `selectedSite?.ID`. All the other components accept 0. Maybe not the `getIsSiteWPCOM` in the header, but there is already a check to handle that.

Reply:
That works, but does that mean we're accepting `0` as a valid `siteId` in the other places? Could that cause issues with the API calls?

Reply:
Since we don't have a 0 site I didn't think it would be an issue but I think you are right. It was just a slightly unsettling to see it used in a couple different ways so I was throwing an idea out. Maybe just forget I said this haha. Or we could fix the types to match haha. Maybe just ignore it though.

Reply:
Yeah it's not great that it's used differently and all over the place. We could follow up and fix the types up the chain to match.